packages/kit/src/core/server/index.js

```diff
@@ -27,13 +25,23 @@ export async function get_server(port, host, use_https, user_config, handler) {
 		}
 	}
 
-	return new Promise((fulfil) => {
-		const server = use_https
+	return Promise.resolve(
+		use_https
 			? https.createServer(/** @type {https.ServerOptions} */ (https_options), handler)
-			: http.createServer(handler);
+			: http.createServer(handler)
+	);
+}
 
+/**
+ * @param {http.Server | https.Server} server
+ * @param {number} port
+ * @param {string} host
+ * @returns {Promise<void>}
+ */
+export async function set_listener(server, port, host) {
+	return new Promise((fulfil) => {
```

Comment:
I might be missing something, but why not just do `server.listen(port, host || '0.0.0.0')`?

Reply:
Listen is asynchronous -- the callback is for waiting until the server is actually listening on that port: [Node Doc](https://nodejs.org/api/net.html#net_server_listen).  The original source used the callback.  I thought that I should, too.

Reply:
It looks like the callback is a no-op if I'm reading it correctly (async isn't my forte, so I could be wrong). If it is indeed a no-op, we should go ahead and remove it regardless of what the code was previously doing

Reply:
Summary: I agree with you.  Change made.

Detail:

The empty callback just ensures that the server is actually listening.  The former `set_listener()` function didn't return until it receives this confirmation.  The new `listener()` function will immediately return to the callee.

Every example of a native node http(s) server or an express server merely displays a console message in the callback, such as 'Server running at http://hostname:port'.

In any web application, browser requests can made at any point in time -- before the server.listen() call, before the server.listen() call is complete, or after completion.  Waiting for the listener to be "ready" doesn't really buy anything, since we have no control over the user's browsing behavior.

Hence, I agree with your suggested change.